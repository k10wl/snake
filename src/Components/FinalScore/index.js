import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

const FinalScore = ({ restart, score }) => (
  <div className="popout">
    <div className="popout--box">
      <p>
        You scored {score} point{score !== 1 && `s`}!
      </p>
      <button onClick={restart} type="button">
        Play again
      </button>
      <Link to="/">
        <button type="button">Main menu</button>
      </Link>
    </div>
  </div>
);

export default FinalScore;

FinalScore.propTypes = {
  restart: PropTypes.func.isRequired,
  score: PropTypes.number.isRequired,
};
