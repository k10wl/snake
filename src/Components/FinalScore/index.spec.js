/* eslint-disable */
import React from "react";
import { shallow } from "enzyme";
import FinalScore from ".";

describe("FinalScore component", () => {
  let props
  beforeEach(() => {
    props = {
      restart: jest.fn(),
      score: 1,
    };
  });
  it(`should return "You scored 1 point!"`, function() {
    let sut = shallow(<FinalScore {...props} />)
    expect(sut.text().includes("You scored 1 point!")).toBe(true);
  });
  it(`should return "You scored 3 points!"`, function(){
    props.score = 3;
    let sut = shallow(<FinalScore {...props} />)
    expect(sut.text().includes("You scored 3 points!")).toBe(true);
  })
  it("should match snapshot", function() {
    let sut = shallow(<FinalScore {...props} />)
    expect(sut).toMatchSnapshot();
  });
});
