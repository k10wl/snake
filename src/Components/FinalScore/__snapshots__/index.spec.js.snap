// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinalScore component should match snapshot 1`] = `
<div
  className="popout"
>
  <div
    className="popout--box"
  >
    <p>
      You scored 
      1
       point
      !
    </p>
    <button
      onClick={[MockFunction]}
      type="button"
    >
      Play again
    </button>
    <Link
      to="/"
    >
      <button
        type="button"
      >
        Main menu
      </button>
    </Link>
  </div>
</div>
`;
